packages/plugin-legacy/src/index.ts

```diff
@@ -255,7 +255,10 @@ function viteLegacyPlugin(options: Options = {}): Plugin[] {
       }
 
       // legacy bundle
-      if (legacyPolyfills.size) {
+      if (legacyPolyfills.size || (!genModern && !options.polyfills)) {
+        // if genModern=false & polyfills=false, we still need to include the
+        // polyfill chunk since it contains SystemJS (see #14324)
```

Comment:
Even if we generate modern chunks, we still need SystemJS for legacy chunks.
So this condition needs to be removed completely.

I guess you want to skip injecting polyfills completely when `options.polyfills === false`. 
For that matter, I think we need to wrap the detectPolyfills below.
```js
if (options.polyfills !== false) {
  // check if the target needs Promise polyfill because SystemJS relies on it
  // https://github.com/systemjs/systemjs#ie11-support
  await detectPolyfills(
    `Promise.resolve(); Promise.all();`,
    targets,
    legacyPolyfills,
  )
}
```

Reply:
I've made this change, should I add a new test case in the `/playground` folder?

Reply:
It'd be great if you could add a test in `playground/legacy`.

Reply:
I've added a test case and confirmed that it fails without this patch